shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -10,21 +10,49 @@
 import android.os.Build;
 import android.os.Bundle;
 import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 import android.util.Log;
 import android.view.View;
 import android.view.accessibility.AccessibilityEvent;
 import android.view.accessibility.AccessibilityNodeInfo;
 import android.view.accessibility.AccessibilityNodeProvider;
 
+import io.flutter.embedding.engine.FlutterJNI;
 import io.flutter.embedding.engine.systemchannels.AccessibilityChannel;
 import io.flutter.util.Predicate;
 
 import java.nio.ByteBuffer;
 import java.util.*;
 
-class AccessibilityBridge
-        extends AccessibilityNodeProvider {
-    private static final String TAG = "FlutterView";
+/**
+ * Bridge between Android's OS accessibility system and Flutter's accessibility system.
+ *
+ * An {@code AccessibilityBridge} requires a real Android {@link View}, called the
+ * {@link #rootAccessibilityView}, which contains a Flutter UI. The {@link #rootAccessibilityView}
+ * is required at the time of {@code AccessibilityBridge}'s instantiation and is held
+ * for the duration of {@code AccessibilityBridge}'s lifespan. {@code AccessibilityBridge}
+ * invokes various accessibility methods on the {@link #rootAccessibilityView}, e.g.,
+ * {@link View#onInitializeAccessibilityNodeInfo(AccessibilityNodeInfo)}. The
+ * {@link #rootAccessibilityView} is expected to notify the {@code AccessibilityBridge}
+ * of relevant interactions: {@link #setAccessibilityEnabled(boolean)},
+ * {@link #handleTouchExploration(float, float)}, {@link #handleTouchExplorationExit()},
+ * and {@link #reset()}.
+ *
+ * The {@code AccessibilityBridge} tricks Android into treating Flutter widgets as if
```

Comment:
I don't really think it should be called a trick, since this is more or less the way any application that renderers to a surface instead of using views is made accessible.

Reply:
Changed the wording.